docker/docker-entrypoint.sh

```diff
@@ -238,4 +238,8 @@ python3 /srv/mapserver/tools/make_mapfile_config.py > /srv/mapserver/sld/config.
 echo Starting server
 # Apache gets grumpy about PID files pre-existing
 rm -f /var/run/apache2/apache2.pid
+# Also remove any socket files that may linger.
+# These may cause EADDRINUSE (Address already in use) if the new Apache's
+# PID matches that of a previous instance.
+rm -f /var/run/apache2/cgisock.*
```

Comment:
I do realize you said this was a hack, but maybe it is worth troubleshooting this further.

Apache is being run from a shell script, namely `docker-entry.sh`, however we are calling `apachectld` directly. This spawns a new process that won't catch any of the signals send to the container. Maybe this is all trivially resolved by doing `exec apachectl -D FOREGROUND`. 

That way the the shell script is replaced by the Apache daemon and now signals send to the Docker container will be received by the Apache daemon. Meaning shutdowns, restart are properly handled by Apache and we don't need to muck around with `rm`ing pid and socket files.

Having said that I don't have much experience with Docker. So maybe there is more to it than I am suggesting.

Reply:
Good point about the signals. It turns out that signals are sent to Apache properly. It's when one of those signals is SIGKILL to the right (/wrong) process that the issue manifests. But it looks like it also manifests with this hack, so I guess I have some more experimenting to do...

Reply:
I think I was looking at a stale image. I can now reliably reproduce the issue by killing the Apache worker processes (but not apachectl) with SIGKILL from outside the container. Removing the socket file fixes the issue in that case.